src/Microsoft.AspNetCore.Razor.Language/Legacy/TokenizerBackedParser.cs

```diff
@@ -570,108 +492,202 @@ protected SyntaxToken AcceptWhiteSpaceInLines()
+                var tokens = new List<SyntaxToken>();
+                do
+                {
+                    if (IsAtEmbeddedTransition(
+                        (mode & BalancingModes.AllowCommentsAndTemplates) == BalancingModes.AllowCommentsAndTemplates,
+                        (mode & BalancingModes.AllowEmbeddedTransitions) == BalancingModes.AllowEmbeddedTransitions))
+                    {
+                        Accept(tokens);
+                        tokens.Clear();
+                        ParseEmbeddedTransition(builder);
+
+                        // Reset backtracking since we've already outputted some spans.
+                        startPosition = CurrentStart.AbsoluteIndex;
+                    }
+                    if (At(left))
+                    {
+                        nesting++;
+                    }
+                    else if (At(right))
+                    {
+                        nesting--;
+                    }
+                    if (nesting > 0)
+                    {
+                        tokens.Add(CurrentToken);
+                    }
+                }
+                while (nesting > 0 && NextToken());
+
+                if (nesting > 0)
+                {
+                    if ((mode & BalancingModes.NoErrorOnFailure) != BalancingModes.NoErrorOnFailure)
+                    {
+                        Context.ErrorSink.OnError(
+                            RazorDiagnosticFactory.CreateParsing_ExpectedCloseBracketBeforeEOF(
+                                new SourceSpan(start, contentLength: 1 /* { OR } */),
+                                Language.GetSample(left),
+                                Language.GetSample(right)));
+                    }
+                    if ((mode & BalancingModes.BacktrackOnFailure) == BalancingModes.BacktrackOnFailure)
+                    {
+                        Context.Source.Position = startPosition;
+                        NextToken();
+                    }
+                    else
+                    {
+                        Accept(tokens);
+                    }
+                }
+                else
+                {
+                    // Accept all the tokens we saw
+                    Accept(tokens);
+                }
             }
-            if (accepts != null)
+            return nesting == 0;
+        }
+
+        protected virtual void ParseEmbeddedTransition(in SyntaxListBuilder<RazorSyntaxNode> builder)
+        {
+        }
+
+        protected internal void AcceptMarkerTokenIfNecessary()
+        {
+            if (TokenBuilder.Count == 0 && Context.LastAcceptedCharacters != AcceptedCharactersInternal.Any)
             {
-                Span.EditHandler.AcceptedCharacters = accepts.Value;
+                Accept(Language.CreateMarkerToken());
             }
         }
 
-        protected virtual void OutputSpanBeforeRazorComment()
+        protected MarkupTextLiteralSyntax OutputAsMarkupLiteral()
```

Comment:
Why are markup output methods in the TokenizeraBackedParser? It possible to move them into the Html parser?

Reply:
Spoke offline. They exist here because the CSharp parser also uses it to parse directives.